src/requests/adapters.py

```diff
@@ -109,6 +113,24 @@ def _urllib3_request_context(
     }
     return host_params, pool_kwargs
 
+def is_single_certificate(cert_path):
```

Comment:
It seems unlikely that special-casing single certs is the best approach 🤔: we probably want to not care about the part of the chain that's above the trusted certificate.
I wouldn't expect curl to special-case single certs either...
Probably we want to match curl's behavior with regards to the openssl config here.